### PR TITLE
Add pxp product units

### PR DESCRIPTION
**Pull request by jenniferboedker**
Adds the new product types for the proteomics platform:

-   PER_HOUR("Hour"),
-   PER_PROJECT("Project"),
-   PER_RUN("Run"),
-   PER_CYCLE("Cycle"),
-   PER_GEL("Gel/HpH"),
-   PER_10_MG("10 milligram"),
-   PER_MEASUREMENT("Measurement"),
-   PER_CHANNEL("Channel"),
-   PER_100_MICROGRAM_PEPTIDE_CHANNEL("100 microgram peptides channel"),
-   PER_500_ML("500 milliliter"),
-   PER_COMPARISON("Comparison")

### CODE REVIEW

**Comment by luiskuhn** [comment]
The modifications to ProductUnit.groovy look straightforward to me

**Review by luiskuhn — DISMISSED** [review]
I think the addition of these new types looks fine

File: src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy [hunk 1]
```diff
@@ -10,7 +10,17 @@ enum ProductUnit {
   PER_GIGABYTE("Gigabyte"),
   PER_SAMPLE("Sample"),
   PER_DATASET("Dataset"),
-  PER_HOUR("Hour")
+  PER_HOUR("Hour"),
+  PER_PROJECT("Project"),
+  PER_RUN("Run"),
+  PER_CYCLE("Cycle"),
+  PER_GEL("Gel/HpH"),
+  PER_MG("10 mg"),
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy:18]: Why is per mg `10 mg` and not `Milligramm` like with Gigabyte?

File: src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy [hunk 2]
```diff
@@ -10,7 +10,17 @@ enum ProductUnit {
...
+  PER_MEASUREMENT("Measurement"),
+  PER_CHANNEL("Channel"),
+  PER_PEPTIDE_CHANNEL("100 µg peptides channel"),
+  PER_ML("500 ml"),
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy:22]: Why is per ml `500 ml` and not `Milliliter` like with Gigabyte?

Comment (jenniferboedker) [reply src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy:22]: Good Catch!

File: src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy [hunk 3]
```diff
@@ -15,11 +15,11 @@ enum ProductUnit {
   PER_RUN("Run"),
   PER_CYCLE("Cycle"),
   PER_GEL("Gel/HpH"),
-  PER_MG("10 mg"),
+  PER_MG("10 Milligram"),
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy:18]:
```suggestion
  PER_10_MG("10 Milligram"),
```

File: src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy [hunk 4]
```diff
@@ -15,11 +15,11 @@ enum ProductUnit {
...
   PER_MEASUREMENT("Measurement"),
   PER_CHANNEL("Channel"),
-  PER_PEPTIDE_CHANNEL("100 µg peptides channel"),
-  PER_ML("500 ml"),
+  PER_PEPTIDE_CHANNEL("100 Microgram peptides channel"),
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy:21]:
```suggestion
  PER_100_MICROGRAM_PEPTIDE_CHANNEL("100 Microgram peptides channel"),
```

File: src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy [hunk 5]
```diff
@@ -15,11 +15,11 @@ enum ProductUnit {
...
+  PER_ML("500 Milliliter"),
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy:22]:
```suggestion
  PER_500_ML("500 Milliliter"),
```

**Review by KochTobi — CHANGES_REQUESTED** [review]

File: src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy [hunk 6]
```diff
@@ -10,7 +10,17 @@ enum ProductUnit {
   PER_GIGABYTE("Gigabyte"),
   PER_SAMPLE("Sample"),
   PER_DATASET("Dataset"),
-  PER_HOUR("Hour")
+  PER_HOUR("Hour"),
+  PER_PROJECT("Project"),
+  PER_RUN("Run"),
+  PER_CYCLE("Cycle"),
+  PER_GEL("Gel/HpH"),
+  PER_MG("10 Milligram"),
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy]: NIT lower case when there is a number in front of it
```suggestion
  PER_MG("10 milligram"),
```

File: src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy [hunk 7]
```diff
@@ -10,7 +10,17 @@ enum ProductUnit {
...
+  PER_MEASUREMENT("Measurement"),
+  PER_CHANNEL("Channel"),
+  PER_PEPTIDE_CHANNEL("100 Microgram peptides channel"),
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy]: NIT 
```suggestion
  PER_PEPTIDE_CHANNEL("100 microgram peptides channel"),
```

File: src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy [hunk 8]
```diff
@@ -10,7 +10,17 @@ enum ProductUnit {
...
+  PER_ML("500 Milliliter"),
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy]:
```suggestion
  PER_ML("500 milliliter"),
```

File: src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy [hunk 9]
```diff
@@ -10,7 +10,17 @@ enum ProductUnit {
   PER_GIGABYTE("Gigabyte"),
   PER_SAMPLE("Sample"),
   PER_DATASET("Dataset"),
-  PER_HOUR("Hour")
+  PER_HOUR("Hour"),
+  PER_PROJECT("Project"),
+  PER_RUN("Run"),
+  PER_CYCLE("Cycle"),
+  PER_GEL("Gel/HpH"),
+  PER_10_MG("10 Milligram"),
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy:18]: NIT lower case when there is a number in front
```suggestion
  PER_10_MG("10 milligram"),
```

File: src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy [hunk 10]
```diff
@@ -10,7 +10,17 @@ enum ProductUnit {
...
+  PER_MEASUREMENT("Measurement"),
+  PER_CHANNEL("Channel"),
+  PER_100_MICROGRAM_PEPTIDE_CHANNEL("100 Microgram peptides channel"),
+  PER_500_ML("500 Milliliter"),
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy:22]:
```suggestion
  PER_500_ML("500 milliliter"),
```

File: src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy [hunk 11]
```diff
@@ -10,7 +10,17 @@ enum ProductUnit {
...
-  PER_HOUR("Hour")
+  PER_HOUR("Hour"),
+  PER_PROJECT("Project"),
+  PER_RUN("Run"),
+  PER_CYCLE("Cycle"),
+  PER_GEL("Gel/HpH"),
+  PER_10_MG("10 Milligram"),
+  PER_MEASUREMENT("Measurement"),
+  PER_CHANNEL("Channel"),
+  PER_100_MICROGRAM_PEPTIDE_CHANNEL("100 Microgram peptides channel"),
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy:21]:
```suggestion
  PER_100_MICROGRAM_PEPTIDE_CHANNEL("100 microgram peptides channel"),
```

File: src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy [hunk 12]
```diff
@@ -10,7 +10,17 @@ enum ProductUnit {
...
+  PER_PROJECT("Project"),
+  PER_RUN("Run"),
+  PER_CYCLE("Cycle"),
+  PER_GEL("Gel/HpH"),
+  PER_10_MG("10 milligram"),
+  PER_MEASUREMENT("Measurement"),
+  PER_CHANNEL("Channel"),
+  PER_100_MICROGRAM_PEPTIDE_CHANNEL("100 microgram peptides channel"),
+  PER_500_ML("500 milliliter"),
+  PER_COMPARISON("COMPARISON")
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/dtos/business/services/ProductUnit.groovy:23]:
```suggestion
  PER_COMPARISON("Comparison")
```

File: CHANGELOG.rst [hunk 13]
```diff
@@ -10,6 +10,9 @@ This project adheres to `Semantic Versioning <https://semver.org/>`_.
 
 **Added**
 
+* Add new product unit types PER_PROJECT, PER_RUN, PER_CYCLE, PER_GEL, PER_MG, PER_MEASUREMENT, PER_CHANNEL, PER_PEPTIDE_CHANNEL, PER_ML, PER_COMPARISON
```
Comment (KochTobi) [inline CHANGELOG.rst:13]:
```suggestion
* Add new product unit types to ``life.qbic.datamodel.dtos.business.services.ProductUnit``
  -   PER_HOUR("Hour"),
  -   PER_PROJECT("Project"),
  -   PER_RUN("Run"),
  -   PER_CYCLE("Cycle"),
  -   PER_GEL("Gel/HpH"),
  -   PER_10_MG("10 milligram"),
  -   PER_MEASUREMENT("Measurement"),
  -   PER_CHANNEL("Channel"),
  -   PER_100_MICROGRAM_PEPTIDE_CHANNEL("100 microgram peptides channel"),
  -   PER_500_ML("500 milliliter"),
  -   PER_COMPARISON("Comparison")
```

File: CHANGELOG.rst [hunk 14]
```diff
@@ -10,6 +10,9 @@ This project adheres to `Semantic Versioning <https://semver.org/>`_.
...
+to 'life.qbic.datamodel.dtos.business.services.ProductUnit' (`#215 <https://github.com/qbicsoftware/data-model-lib/pull/215>`_)
```
Comment (KochTobi) [inline CHANGELOG.rst:14]:
```suggestion
```

Comment (jenniferboedker) [reply CHANGELOG.rst:14]: I will resolve this. I accept the first comment.

Comment (KochTobi) [reply CHANGELOG.rst:26]: was moved in previous suggestion

**Review by KochTobi — APPROVED** [review]
LGTM

**Comment by jenniferboedker** [comment]
We will wait with merging until we have the final list of units

**Pull request merged**